Вторая домашняя работа

olerom opened the pull request.

Commit:
Add todos.css, icons and bottom-cards for index.html

Commit:
Add font-style

Commit:
Refactor project structure: move to MCSS

Commit:
Remove fonts, fix checkbox and remove button position

DmitryMakhnev commented on index.html:
```diff
@@ -15,28 +31,37 @@ <h1 class="todos-title">todos</h1>
 <div class="todos-list">
     <div class="todo-item">
-        <textarea class="todo-item_text">HTML</textarea>
+        <!--Нельзя ли заменить чекбокс на div?-->
```
В разборе ДЗ описано

DmitryMakhnev commented on styles/base/button/filter-button.css:
```diff
@@ -0,0 +1,13 @@
+    cursor: pointer;
+}
+
+.__active {
```
Нельзя писать стили на модификатор без указания контекста использования.
Проблема в том, что если я добавлю такой же модификатор какому либо другому элементу (а элементов с модификатором `__active` может быть очень много), этот элемент получит стили описанные здесь.

DmitryMakhnev commented on styles/base/checkbox/checkbox.css:
```diff
@@ -0,0 +1,29 @@
+input[type=checkbox] {
```
В проекте может быть n чекбокосв.
Надо делать компонент по классу

DmitryMakhnev commented on styles/base/user-input/item-text.css:
```diff
@@ -0,0 +1,20 @@
+    top: 6px;
+}
+
+.__ready .todo-item_text {
```
Та же проблема с модификатором

DmitryMakhnev commented on styles/project/add-field/todos-add.css:
```diff
@@ -0,0 +1,7 @@
+.todos-add {
+    border: 1px solid #e6e6e6;
+    width: 600px;
```
Нужно стараться делать всё резиновым

DmitryMakhnev reviewed (commented):
Синхронизировать с разбором ДЗ

Commit:
Fix mistakes:
-remove fixed sizes
-add correct input checkboxes
-add svg images
-add full modifiers

Commit:
Add animation for complete-checkbox

Commit:
Move to SASS

Commit:
Add color variables, correct width properties

Commit:
Add mixins and varibles: offsets, fonts, layout

DmitryMakhnev commented on package.json:
```diff
@@ -0,0 +1,25 @@
+  },
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:ok-technopolis/preprocessor-stub.git"
```
Наверное стоит поправить?

DmitryMakhnev commented on public/index.html:
```diff
@@ -0,0 +1,111 @@
+<div class="wrapper">
+    <div class="todos-add">
+        <div class="todo-add_input-checkbox">
+            <input type="checkbox"
```
Это не чекбокс, а экшен

DmitryMakhnev commented on src/styles/base/button/clear-completed-button.scss:
```diff
@@ -0,0 +1,12 @@
+@import "../../config/globals";
+
+.todos-actions-bar_clear-completed {
```
1) Разве это базовый компонент?
2) В `.todos-actions-bar_clear-completed` блоком является `todos-actions-bar`
`todos-actions-bar_clear-completed` не может быть отдельным компонентом
3) `clear-completed-button` !== `todos-actions-bar_clear-completed`

DmitryMakhnev commented on src/styles/base/button/clear-completed-button.scss:
```diff
@@ -0,0 +1,12 @@
+    float: right;
+    outline: none;
+    cursor: pointer;
+    order: 3;
```
Флексы там не нужны, смотри разбор дз

DmitryMakhnev commented on src/styles/base/button/filter-button.scss:
```diff
@@ -0,0 +1,20 @@
+@import "../../config/globals";
+
+.todos-filter,
+.todos-actions-bar_clear-completed {
```
Почему в одном компоненте описаны стили для другого? O_o

DmitryMakhnev commented on src/styles/base/button/filter-button.scss:
```diff
@@ -0,0 +1,20 @@
+    margin-right: $offset-main-content;
+}
+
+.todos-filters .todos-filter.__active {
```
Зачем кскадность от `.todos-filters`

DmitryMakhnev commented on src/styles/base/button/remove-button.scss:
```diff
@@ -0,0 +1,21 @@
+@import "../../config/globals";
+
+.todo-item_remove {
+    background: inherit;
```
Зачем?

DmitryMakhnev commented on src/styles/base/button/remove-button.scss:
```diff
@@ -0,0 +1,21 @@
+@import "../../config/globals";
```
remove-button !== todo-item_remove

DmitryMakhnev commented on src/styles/base/button/remove-button.scss:
```diff
@@ -0,0 +1,21 @@
+    float: right;
+    outline: none;
+    cursor: pointer;
+    position: absolute;
```
float + absolute + flex 🌈
1) наверное что-то лишнее
2) см https://github.com/ok-technopolis/Shvedov_Oleg/pull/4#discussion_r115477715

DmitryMakhnev commented on src/styles/base/checkbox/list-item-checkbox.scss:
```diff
@@ -0,0 +1,76 @@
+    transform: rotate(37deg);
+    transition: transform .2s;
+
+    margin-top: 7.4px;
```
дробные пиксели? O_o

DmitryMakhnev commented on src/styles/base/checkbox/list-item-checkbox.scss:
```diff
@@ -0,0 +1,76 @@
+    height: 100%;
+    background: url('../../../images/svg/t2.svg') no-repeat left center;
+
+    transform: rotate(37deg);
```
Картинка + трансформ... Очень не понятно.
Почему нельзя было делать одной картинкой?
Зачем экстра элементы?

olerom replied:
Получилось таким костыльным способом сделать анимацию.
Убрал

DmitryMakhnev commented on src/styles/base/checkbox/select-all-checkbox.scss:
```diff
@@ -0,0 +1,41 @@
+@import "../../config/globals";
+
+.todo-add_input-checkbox {
```
Этот компонент должен быть не чекбоксом

DmitryMakhnev commented on src/styles/base/default.scss:
```diff
@@ -0,0 +1,18 @@
+    margin: 0 0 0 0;
+}
+
+h1 {
```
Зачем?

DmitryMakhnev commented on src/styles/base/user-input/add-input.scss:
```diff
@@ -0,0 +1,15 @@
+
+.todo-add_input {
+    border: none;
+    background: $c-main-content-bg !important; /* желательно прописывать, так как в Chrome при сохранных данных оно может быть желтым */
```
`!important` в данном случае не поможет, по крайней мере в тех версиях что я трогал.
Но в данном месте не стоит на это загоняться, тут не будет честного сабмита.

DmitryMakhnev commented on src/styles/project/actions-bar/actions-bar.scss:
```diff
@@ -0,0 +1,27 @@
+.todos-actions-bar_counter {
+    max-width: 33%;
+    float: left;
+    order: 1;
```
Здесь не нужен флекс

DmitryMakhnev reviewed (commented):
Очень много проблем, рекомендуется пересмотреть материалы занятий по вёрстке и изучить разбор дз

Commit:
Fix todos-add problems

Commit:
Fix todos-actions-bar: remove wrong component placing for clear-completed

Commit:
Fix todos-list problems

Commit:
Remove redundant position attribute from todos-item

Commit:
Correct component placing

DmitryMakhnev commented on src/styles/project/todo-list/todo-item.scss:
```diff
@@ -0,0 +1,67 @@
+        margin-top: $offset_3x;
+        margin-bottom: $offset_3x;
+        order: 1;
+        align-self: flex-end;
```
Должен быть фаллбек на флоатах, который работает

DmitryMakhnev commented on src/styles/project/todos-actions-bar/todos-actions-bar.scss:
```diff
@@ -0,0 +1,38 @@
+}
+
+    .todos-actions-bar_counter {
+        max-width: 33%;
```
Не должно быть указано ширины. Посмотрите разбор домашнего задания

DmitryMakhnev reviewed (commented):
Пофиксить комменты

Commit:
Fix mistakes: fallback for flex

DmitryMakhnev commented on src/styles/base/filter/filter.scss:
```diff
@@ -0,0 +1,14 @@
+}
+
+.filter.__active {
+    border: 1px solid $c-default-border;
```
от таких правил будут происходить прыжки

DmitryMakhnev commented on src/styles/base/filter/filter.scss:
```diff
@@ -0,0 +1,14 @@
+    background-color: $c-filters-background;
+    color: $c-text;
+    outline: none;
+    cursor: pointer;
```
По дизайну были паддинги

DmitryMakhnev commented on src/styles/base/input-checkbox/input-checkbox.scss:
```diff
@@ -0,0 +1,43 @@
+
+        .input-checkbox_target:checked ~ .input-checkbox_visual {
+            border: 1px solid $c-checkbox-active;
+            background: url('../../../images/svg/completed.svg') no-repeat center;
```
Для свг тоже могут использовать спрайты, поэтому лучше положить иконку согласно нужному размеру

DmitryMakhnev commented on src/styles/project/todo-list/todo-item.scss:
```diff
@@ -0,0 +1,67 @@
+            word-wrap: break-word;
+            align-self: center;
+            word-break: break-all;
+            flex: 1 0;
```
флекс же на обёртке

DmitryMakhnev commented on src/styles/project/todos-add/todos-add.scss:
```diff
@@ -0,0 +1,66 @@
+    .todo-add_select-all-action .action_visual {
+        width: 24px;
+        height: 12px;
+        overflow: hidden;
```
Зачем?

DmitryMakhnev reviewed (approved):
Поправить мелкие недочёты

Commit:
Fix mistakes: flex for item, remove redundant overflow hidden for todos-add
